### PR TITLE
User pagination

### PR DESCRIPTION
**Pull request by chien123-exam**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use App\Http\Requests\SaveUserRequest;
+use App\Models\Family;
+use Illuminate\Support\Facades\Storage;
+
+class UserController extends Controller
+{
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php:12]: Thêm cho anh method __construct()
Bên trong sẽ injection 2 model Family và User vào như sau

   ```
 protected $familyModel;
    protected $userModel;
    public function __construct(Family $family, User $user)
    {
        $this->familyModel = $family;
        $this->userModel = $user;
    }

```

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    public function index(Request $request)
+    {
+        $inputs = $request->all();
+
+        $query = User::query();
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: User::
=>
$this->userModel

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    public function create()
+    {
+        return view('users.form', [
+            'families' => Family::all(),
+        ]);
+    }
+
+    public function store(SaveUserRequest $request)
+    {
+        $inputs =$request->all();
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: $inputs =$request->all();
=>
$inputs = $request->all();

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        $inputs['password'] = bcrypt($request->password);
+        $inputs['type'] = User::TYPE['admin'];
+
+        if ($request->avatar) {
+            $inputs['avatar'] = Storage::disk('public')->put('media', $request->avatar);
+        }
+
+        User::create($request->all());
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: User::
=>
$this->userModel

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    }
+
+    public function edit($id)
+    {
+        return view('users.form', [
+            'user' => User::find($id),
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: User::
=>
$this->userModel

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            'families' => Family::all(),
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: Family::
=>
$this->familyModel

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        ]);
+    }
+
+
+    public function update(SaveUserRequest $request, $id)
+    {
+        // dd($request->all());
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: Remove it

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        $inputs = array_filter($request->all());
+
+        if($request->password) {
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: if($request->password) {
=>
 if ($request->password) {

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+            $inputs['password'] = bcrypt($request->password);
+        }
+
+        if ($request->avatar) {
+            $inputs['avatar'] = Storage::disk('public')->put('media', $request->avatar);
+        }
+
+        User::find($id)->update($inputs);
```
Comment (phamkykhoi) [inline app/Http/Controllers/UserController.php]: User::
=>
$this->userModel

File: app/Http/Requests/SaveUserRequest.php [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            $rules['password_confirm'] = ['required', 'same:password'];
+        }
+
+        //When update
+
+        if (!empty($this->user)) {
+            $rules['password'] = ['nullable', 'min:6'];
+            $rules['password_confirm'] = ['nullable', 'same:password'];
+        }
+
```
Comment (phamkykhoi) [inline app/Http/Requests/SaveUserRequest.php:48]: Remove empty line

File: app/Models/User.php [hunk 11]
```diff
@@ -43,4 +61,31 @@ class User extends Authenticatable
...
+        if ($this->attributes['gender'] == static::GENDER['famale']) {
+            return 'Famale';
+        }
+
+        return null;
+    }
+
+    public function getUserTypeAttribute()
+    {
+        if( $this->attributes['type'] == static::TYPE['admin']) {
```
Comment (phamkykhoi) [inline app/Models/User.php]: if(
=>
 if (

File: database/seeders/DatabaseSeeder.php [hunk 12]
```diff
@@ -12,16 +12,17 @@ class DatabaseSeeder extends Seeder
...
         // \App\Models\User::factory()->create([
         //     'name' => 'Test User',
         //     'email' => 'test@example.com',
         // ]);
 
-        $this::call([
-            CategorySeeder::class,
-            CourseSeeder::class
-        ]);
+        // $this::call([
```
Comment (phamkykhoi) [inline database/seeders/DatabaseSeeder.php]: Remove comment

File: database/seeders/DatabaseSeeder.php [hunk 13]
```diff
@@ -12,16 +12,17 @@ class DatabaseSeeder extends Seeder
      */
     public function run(): void
     {
-        // \App\Models\User::factory(10)->create();
+        \App\Models\User::factory(100)->create();
 
         // \App\Models\User::factory()->create([
         //     'name' => 'Test User',
```
Comment (phamkykhoi) [inline database/seeders/DatabaseSeeder.php]: Remove comment

File: database/seeders/DatabaseSeeder.php [hunk 14]
```diff
@@ -12,16 +12,17 @@ class DatabaseSeeder extends Seeder
      */
     public function run(): void
     {
-        // \App\Models\User::factory(10)->create();
```
Comment (phamkykhoi) [inline database/seeders/DatabaseSeeder.php:15]: Remove comment

File: app/Http/Controllers/UserController.php (hunk 1 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php:12]: fixed

File: app/Http/Controllers/UserController.php (hunk 2 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 4 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 5 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 6 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 7 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 8 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Controllers/UserController.php (hunk 9 above)
Comment (chien123-exam) [reply app/Http/Controllers/UserController.php]: fixed

File: app/Http/Requests/SaveUserRequest.php (hunk 10 above)
Comment (chien123-exam) [reply app/Http/Requests/SaveUserRequest.php:48]: fixed

File: app/Models/User.php (hunk 11 above)
Comment (chien123-exam) [reply app/Models/User.php]: fixed

File: database/seeders/DatabaseSeeder.php (hunk 12 above)
Comment (chien123-exam) [reply database/seeders/DatabaseSeeder.php]: fixed

File: database/seeders/DatabaseSeeder.php (hunk 13 above)
Comment (chien123-exam) [reply database/seeders/DatabaseSeeder.php]: fixed

File: database/seeders/DatabaseSeeder.php (hunk 14 above)
Comment (chien123-exam) [reply database/seeders/DatabaseSeeder.php:15]: fixed

**Pull request merged**